### ISSUE TITLE
proxy_connect_allow all ports?

### ISSUE THREAD

**Issue by platkaw** [chobits/ngx_http_proxy_connect_module#4]
Hi, I'm using the module for FTP proxy, but it doesn't work because of the FTP passive mode. Would it be possible to enable the proxy_connect_allow option to accept all ports or range of ports (ex. 1000-2000)? Thank you.

I am using curl with the following options:
```
curl -p -k -v --globoff --ssl -u ${USERNAME}:${PASSWORD} \
     --disable-epsv \
     --proxy $PROXY \
     --connect-timeout 60 \
     --retry 60 \
     --retry-delay 5 \
     --retry-max-time 60 -T ${FILE} ftp://ftp.userscloud.com:21 
```
Below is a sample output:
```
* Hostname was NOT found in DNS cache
*   Trying 10.10.10.1...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to 10.10.10.1 (10.10.10.1) port 8888 (#0)
* Establish HTTP proxy tunnel to ftp.userscloud.com:21
* Server auth using Basic with user 'user'
> CONNECT ftp.userscloud.com:21 HTTP/1.1
> Host: ftp.userscloud.com:21
> User-Agent: curl/7.38.0
> Proxy-Connection: Keep-Alive
> 
< HTTP/1.0 200 Connection Established
< Proxy-agent: nginx
< 
* Proxy replied OK to CONNECT request
< 220 ProFTPD 1.3.4b Server (ftp.userscloud.com) [89.238.134.150]
> AUTH SSL
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0< 234 AUTH SSL successful
* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
* SSLv3, TLS handshake, Client hello (1):
} [data not shown]
* SSLv3, TLS handshake, Server hello (2):
{ [data not shown]
* SSLv3, TLS handshake, CERT (11):
{ [data not shown]
* SSLv3, TLS handshake, Server key exchange (12):
{ [data not shown]
* SSLv3, TLS handshake, Request CERT (13):
{ [data not shown]
* SSLv3, TLS handshake, Server finished (14):
{ [data not shown]
* SSLv3, TLS handshake, CERT (11):
} [data not shown]
* SSLv3, TLS handshake, Client key exchange (16):
} [data not shown]
* SSLv3, TLS change cipher, Client hello (1):
} [data not shown]
* SSLv3, TLS handshake, Finished (20):
} [data not shown]
* SSLv3, TLS change cipher, Client hello (1):
{ [data not shown]
* SSLv3, TLS handshake, Finished (20):
{ [data not shown]
* SSL connection using TLSv1.2 / DHE-RSA-AES256-GCM-SHA384
* Server certificate:
* 	 subject: OU=Domain Control Validated; OU=EssentialSSL Wildcard; CN=*.userscloud.com
* 	 start date: 2015-07-01 00:00:00 GMT
* 	 expire date: 2017-07-09 23:59:59 GMT
* 	 issuer: C=GB; ST=Greater Manchester; L=Salford; O=COMODO CA Limited; CN=COMODO RSA Domain Validation Secure Server CA
* 	 SSL certificate verify result: certificate has expired (10), continuing anyway.
> USER user
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0< 331 Password required for user
> PASS password
< 230 User user logged in
> PBSZ 0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0< 200 PBSZ 0 successful
> PROT P
< 200 Protection set to Private
> PWD
< 257 "/" is the current directory
* Entry path is '/'
  0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0> PASV
* Connect data stream passively
* ftp_perform ends with SECONDARY: 0
< 227 Entering Passive Mode (89,238,134,150,150,119).
* Hostname was found in DNS cache
*   Trying 10.10.10.1...
* Connecting to 89.238.134.150 (10.10.10.1) port 8888
* Connected to 10.10.10.1 (10.10.10.1) port 8888 (#0)
* Connection to proxy confirmed
* Establish HTTP proxy tunnel to 89.238.134.150:38519
* Server auth using Basic with user 'user'
> CONNECT 89.238.134.150:38519 HTTP/1.1
> Host: 89.238.134.150:38519
> User-Agent: curl/7.38.0
> Proxy-Connection: Keep-Alive
> 
> TYPE I
< 200 Type set to I
> STOR file
  0     0    0     0    0     0      0      0 --:--:--  0:00:13 --:--:--     0* response reading failed
* Closing connection 0
* SSLv3, TLS alert, Client hello (1):
} [data not shown]
curl: (56) response reading failed
```

**Comment by chobits** [comment]
hi @platkaw,
 Good question! It's inconvenient to specify every port in configure.
 I will support function that allows all ports or range of ports.

**Comment by chobits** [comment]
I have fixed this issue in pr: https://github.com/chobits/ngx_http_proxy_connect_module/pull/5.

After reviewing by my collaborator, it will be merged soon.

**Comment by chobits** [comment]
hi @platkaw 

This issue has been fixed in pr: #5.

Use the following configure for allowing range of port:

```
proxy_connect_allow 1000-2000;
```

or allowing all ports:

```
proxy_connect_allow all;
```
